modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -112,6 +114,22 @@ public ObjectEntry addObjectEntry(
 			String scopeKey)
 		throws Exception {
 
+		ServiceContext serviceContext = new ServiceContext();
```

Comment:
Hey @seltong, please extract this code to a method and reuse it here and in the update method.

Reply:
I think you can simplify a little using the below version:

```
		Map<String, Object> properties = objectEntry.getProperties();

		serviceContext.setAssetCategoryIds(
			ListUtil.toLongArray(
				(List<String>)properties.get("categoryIds"), Long::parseLong));
		serviceContext.setAssetTagNames(
			ArrayUtil.toStringArray((List<String>)properties.get("tagNames")));

		serviceContext.setUserId(dtoConverterContext.getUserId());
```

Thx

Reply:
Some points related to the code above:

1. I removed the casts for ArrayList since it is not needed and it is always better do not to depend on implementations
2. I replace GetterUtil::getLong with Long::parseLong because the getLong is more expensive since it handles some conversion cases that probably you would not get here.

Thx again

Reply:
It makes sense

Thx